tests/supervisors/test_reload.py

```diff
@@ -36,63 +39,354 @@ def _reload_tester(self, reloader: BaseReload, file: Path) -> bool:
+            config = Config(app="tests.test_config:asgi_app", reload=True)
+            reloader = self._setup_reloader(config)
+
+            assert not self._reload_tester(reloader, file)
+
+            reloader.shutdown()
 
-        update_file = self.tmp_path.joinpath(file)
-        update_file.touch()
+    @pytest.mark.parametrize("reloader_class", [StatReload, WatchGodReload])
+    def test_should_reload_when_directories_have_same_prefix(self) -> None:
+        app_dir = self.reload_path / "app"
+        app_file = app_dir / "src" / "main.py"
+        app_first_dir = self.reload_path / "app_first"
+        app_first_file = app_first_dir / "src" / "main.py"
 
-        with self.tmpdir.as_cwd():
-            config = Config(app=None, reload=True)
+        with as_cwd(self.reload_path):
+            config = Config(
+                app="tests.test_config:asgi_app",
+                reload=True,
+                reload_dirs=[str(app_dir), str(app_first_dir)],
+            )
             reloader = self._setup_reloader(config)
 
-            assert not self._reload_tester(reloader, update_file)
+            assert self._reload_tester(reloader, app_file)
+            assert self._reload_tester(reloader, app_first_file)
 
             reloader.shutdown()
 
-    def test_should_reload_when_directories_have_same_prefix(self):
-        file = "example.py"
+    @pytest.mark.parametrize("reloader_class", [StatReload, WatchGodReload])
+    def test_should_not_reload_when_only_subdirectory_is_watched(self) -> None:
+        app_dir = self.reload_path / "app"
+        app_dir_file = self.reload_path / "app" / "src" / "main.py"
+        root_file = self.reload_path / "main.py"
+
+        with as_cwd(self.reload_path):
+            config = Config(
+                app="tests.test_config:asgi_app",
+                reload=True,
+                reload_dirs=[str(app_dir)],
+            )
+            reloader = self._setup_reloader(config)
+
+            assert self._reload_tester(reloader, app_dir_file)
+            assert not self._reload_tester(reloader, root_file)
+
+            reloader.shutdown()
+
+    @pytest.mark.parametrize("reloader_class", [StatReload, WatchGodReload])
+    def test_should_parse_dir_from_includes(self) -> None:
+        app_dir = self.reload_path / "app"
+        app_file = app_dir / "src" / "main.py"
+        app_first_dir = self.reload_path / "app_first"
+        app_first_file = app_first_dir / "src" / "main.py"
+
+        with as_cwd(self.reload_path):
+            config = Config(
+                app="tests.test_config:asgi_app",
+                reload=True,
+                reload_includes=[str(app_dir)],
+            )
+            reloader = self._setup_reloader(config)
+
+            assert self._reload_tester(reloader, app_file)
+            assert not self._reload_tester(reloader, app_first_file)
+
+            reloader.shutdown()
+
+    @pytest.mark.parametrize("reloader_class", [StatReload])
```

Comment:
ok here I dont follow, I guess  that will be different here for the `WatchGodReload` case ? why not doing the same as in `test_reload_when_pattern_matched_file_is_changed` or put differently, what does this test differently than `test_reload_when_pattern_matched_file_is_changed`

Reply:
Sorry Github is not showing the test name, which one is this comment for?

Reply:
for     `def test_should_not_parse_filetype_from_includes(self) -> None:`

Reply:
Ok, it's an old duplicate test. I have removed it